### ISSUE TITLE
matmul doesn't preserve the batch dimension with size 1 

### ISSUE THREAD

**Issue by huningxin** [tensorflow/tfjs#4192]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow.js):
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): macOS 10.15.7
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow.js installed from (npm or script link): <script src="https://cdn.jsdelivr.net/npm/@tensorflow/tfjs@2.7.0"> </script>
- TensorFlow.js version (use command below): 2.7.0
- Browser version: Chrome 86.0.4240.111
- Tensorflow.js Converter Version:

**Describe the current behavior**
a = tf.ones([1, 2, 3]);
b = tf.ones([3, 4]);
c = tf.matmul(a, b);
c.shape; // [2, 4]

**Describe the expected behavior**
c.shape should be [1, 2, 4]

**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/CodePen/any notebook.

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

**Comment by tafsiri** [comment]
Could you tell us more about how this situation came about? While it does work in python tf according to the docs for the [python api](https://www.tensorflow.org/api_docs/python/tf/linalg/matmul)

'The inputs must, following any transpositions, be tensors of rank >= 2 where the inner 2 dimensions specify valid matrix multiplication dimensions, and any further outer dimensions specify matching batch size.'

So it seems by the docs that outer dimensions should always match, did this emerge from trying to run a model?

**Comment by huningxin** [comment]
> While it does work in python tf according to the docs for the [python api](https://www.tensorflow.org/api_docs/python/tf/linalg/matmul)

Python API works as expected:
```
>>> tf.version.VERSION
'2.3.1'
>>> a = tf.ones([1, 2, 3])
>>> b = tf.ones([3, 4])
>>> c = tf.matmul(a, b)
>>> c.shape
TensorShape([1, 2, 4])
```

**Comment by tafsiri** [comment]
Yes it does work, but that may be more by accident than design. Could you let us know how more about how you came across this issue, that will help us in prioritization.

**Comment by huningxin** [comment]
Sorry for late response as I was on vacation.

I found this issue by using TF.js to implement the [matmul](https://webmachinelearning.github.io/webnn/#api-modelbuilder-matmul) op of WebNN in its [polyfill](https://github.com/webmachinelearning/webnn-polyfill). The matmul of WebNN is compatible to matmul of TF. So if TF.js follows the same behavior, it would simplify the webnn-polyfill implementation.
